### PR TITLE
Add ability to schedule injector pod in Fargate

### PR DESCRIPTION
**Pull request by isimluk**

### CODE REVIEW

File: apis/falcon/v1alpha1/falconcontainer_types.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ type RegistrySpec struct {
 	TLS RegistryTLSSpec `json:"tls,omitempty"`
 	// Azure Container Registry Name represents the name of the ACR for the Falcon Container push. Only applicable to Azure cloud.
 	AcrName *string `json:"acr_name,omitempty"`
+	// EcrIamRoleArnForInjector allows injector to assume AIM role to access registry
+	EcrIamRoleArnForInjector *string `json:"ecr_iam_role_arn_for_injector,omitempty"`
```
Comment (redhatrises) [inline apis/falcon/v1alpha1/falconcontainer_types.go:56]: Can we make the json name easier e.g. `ecr_iam_arn`

Comment (isimluk) [reply apis/falcon/v1alpha1/falconcontainer_types.go:56]: what about `ecr_iam_role_arn`

There is a difference between role arn and policy arn and users will definitely stumble upon this.

**Comment by isimluk** [comment]
@RutledgePaulV, I think I am done. Can You please take a brief look at the documentation for this and let me know if you see some easy wins there?

https://github.com/isimluk/falcon-operator/blob/eks-fargate/docs/deployment/eks-fargate/README.md

File: apis/falcon/v1alpha1/falconcontainer_types.go (hunk 1 above)
Comment (redhatrises) [reply apis/falcon/v1alpha1/falconcontainer_types.go]: Sounds good to me.

Comment (isimluk) [reply apis/falcon/v1alpha1/falconcontainer_types.go]: Thank You from me.

Comment (redhatrises) [reply apis/falcon/v1alpha1/falconcontainer_types.go]: hahaha

**Comment by isimluk** [comment]
Thank You @RutledgePaulV for the review and collaboration on this!

**Pull request merged**